### PR TITLE
(ood-gen) Make optional list types in 'tutorials' type non-optional

### LINKED ISSUE #1970: Replace `list option` by `list` in data
Data served by ocaml.org is static. Yaml files' headers from the `/data` folder are parsed and linked into the binary. The `ood-gen` tool turns those files into ocaml sources compiled with the rest of the project.

The Yaml syntax is implicitly described by the `metadata` types in the files from the `tool/ood-gen/lib` folder. Parsing returns values of this type. However, at run time, another type is used. It is the type named `t` in the same file.

Some Yaml headers have optional fields containing lists of values. Their parsers produce values of type `list option`; this is not optimal. It adds unnecessary processing of `Some []` cases. 
blob/main/tool/ood-gen/lib/exercise.ml
This issue is about changing `list option` typed fields in `t` types of the following files:
- [ ] [`tool/ood-gen/lib/changelog.ml`](../blob/main/tool/ood-gen/lib/changelog.ml)
- [ ] [`tool/ood-gen/lib/exercise.ml`](../blob/main/tool/ood-gen/lib/exercise.ml) (#2039)
- [ ] [`tool/ood-gen/lib/news.ml`](../blob/main/tool/ood-gen/lib/news.ml)
- [ ] [`tool/ood-gen/lib/planet.ml`](../blob/main/tool/ood-gen/lib/planet.ml)
- [ ] [`tool/ood-gen/lib/workshop.ml`](../blob/main/tool/ood-gen/lib/workshop.ml/)

The key idea is to replace values `Some []` by `[]`. PR #1967 contains an example of such a change of type. Note that each change of type affects others files and that the types `metadata` should not be changed.

### PR DESCRIPTION
**Pull request by egmaleta**
~~Closes~~ One task from #1970.
The only file listed in the issue with a `option` type in type `t` is `tools/ood-gen/lib/exercise.ml`.
My first PR :).

### CODE REVIEW

**Review by cuihtlauac — APPROVED** [review]
Muchas gracias, @egmaleta ;-) Nice first PR.

This looks good to me, however it does not fix #1970, only one task.

**Comment by egmaleta** [comment]
Hi @cuihtlauac, there are two cases I didn't handle because I didn't think they met the criteria of the issue:

1- `authors` field in type `t` of `Post` module (in `planet.ml`)
2- `additional_links` field in type `presentation` (in `workshop.ml`)

shoul I handle them? If I missed other cases please show me so I can work on them too :)

**Comment by egmaleta** [comment]
another question: to address the other tasks, should I keep commiting changes in my fork's branch or make another pr?

**Comment by cuihtlauac** [comment]
Make another pr, easy review, fast merge

File: tool/ood-gen/lib/exercise.ml [hunk 1]
```diff
@@ -47,12 +47,14 @@ type t = {
   description : string;
   statement : string;
   solution : string;
-  tutorials : string list option;
+  tutorials : string list;
 }
 [@@deriving
-  stable_record ~version:metadata ~remove:[ statement; solution ],
+  stable_record ~version:metadata ~modify:[ tutorials ]
+    ~remove:[ statement; solution ],
```
Comment (cuihtlauac) [inline tool/ood-gen/lib/exercise.ml]:
```suggestion
  stable_record ~version:metadata ~remove:[ statement; solution ]
     ~modify:[ tutorials ],
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]

**Comment by cuihtlauac** [comment]
This looks good to me. Thanks @egmaleta !

**Comment by egmaleta** [comment]
glad to be helpful

**Pull request merged**